exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -15,97 +15,134 @@
 package splunkhecexporter
 
 import (
-	"errors"
 	"fmt"
 	"math"
-	"strconv"
 
-	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/translator/conventions"
-	"go.opentelemetry.io/collector/translator/internaldata"
 	"go.uber.org/zap"
-	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk"
 )
 
 const (
 	// unknownHostName is the default host name when no hostname label is passed.
 	unknownHostName = "unknown"
-	// separator for metric values.
-	separator = "."
 	// splunkMetricValue is the splunk metric value prefix.
 	splunkMetricValue = "metric_name"
-	// bucketSuffix is the bucket suffix for distribution buckets.
-	bucketSuffix = "bucket"
-	// quantileSuffix is the quantile suffix for summary quantiles.
-	quantileSuffix = "quantile"
 	// countSuffix is the count metric value suffix.
 	countSuffix = "count"
-	// sumOfSquaredDeviation is the sum of squared deviation metric value suffix.
-	sumOfSquaredDeviation = "sum_of_squared_deviation"
-)
-
-var (
-	// infinity bound dimension value is used on all histograms.
-	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
 )
 
 func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config) ([]*splunk.Event, int, error) {
-	ocmds := internaldata.MetricsToOC(data)
 	numDroppedTimeSeries := 0
-	_, numPoints := data.MetricAndDataPointCount()
-	splunkMetrics := make([]*splunk.Event, 0, numPoints)
-	for _, ocmd := range ocmds {
-		var host string
-		if ocmd.Resource != nil {
-			host = ocmd.Resource.Labels[conventions.AttributeHostHostname]
+	var splunkMetrics []*splunk.Event
```

Comment:
Pre-allocate the slice:
```golang
_, dpCount := data.MetricAndDataPointCount()
splunkMetrics := make([]*splunk.Event, 0, dpCount)
```

And if you want more optimal:
```
splunkMetricsOrigs := make([]splunk.Event, dpCount)
splunkMetrics := make([]*splunk.Event, 0, dpCount)
...
  append(splunkMetrics, &splunkMetricsOrigs[i])
  // change the createEvent to fillEvent and avoid allocation.
  fillEvent(dataPt.Timestamp(), host, source, sourceType, index, fields, &splunkMetricsOrigs[i])
...
```

Reply:
that's probably not a good idea - in the case of a histogram, we will generate additional points.

Reply:
I know that's not what the code says now, but that's the next job - mapping better to histogram for Splunk HEC.

Reply:
And? Still batch enough allocations, also the first suggestion should be done anyway.

Reply:
You're right I'll get on it.

Reply:
OK, I've done the first part.